ocean_lib/models/datatoken_base.py

```diff
@@ -287,16 +293,22 @@ def create_exchange(
         return (exchange, tx) if kwargs.get("full_info") else exchange
 
     @enforce_types
-    def get_exchanges(self) -> list:
+    def get_exchanges(self, active=True) -> list:
         """return List[OneExchange] - all the exchanges for this datatoken"""
         # import now, to avoid circular import
         from ocean_lib.models.fixed_rate_exchange import OneExchange
 
-        FRE = self._FRE()
+        # import now, to avoid circular import
+        from ocean_lib.models.fixed_rate_exchange import FixedRateExchange
+
+        exchanges = []
         addrs_and_exchange_ids = self.getFixedRates()
-        exchanges = [
-            OneExchange(FRE, exchange_id) for _, exchange_id in addrs_and_exchange_ids
-        ]
+        for address, exchange_id in addrs_and_exchange_ids:
+            fre = FixedRateExchange(self.config_dict, address)
+            exchange = OneExchange(fre, exchange_id)
+            if active and not exchange.is_active():
+                continue
+            exchanges.append(exchange)
```

Comment:
Slightly cleaner, because two lines and avoids negative:
```python
            if active and exchange.is_active():
                exchanges.append(exchange)
```